go/cmd/vtgate/plugin_prombackend.go

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright 2017 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+// This plugin imports Prometheus to allow for instrumentation
+// with the Prometheus client library
+
+import (
+	"vitess.io/vitess/go/stats/prombackend"
+)
+
+func init() {
+	prombackend.Init("vtgate")
```

Comment:
Starting a thread here instead of using Reviewable:

I wrote:

> Is it necessary to have a different namespace for each binary?
> If so, what about other binaries which we have e.g. vtctld and vtworker?
> Does Prometheus require that? Its internal Google counterpart doesn't care and instead can distinguish binaries by their production job name ;)

zmagg@ responded:

> @demmer and I talked about this a while back, but I forget why we landed on this. Prometheus does let you specify a namespace per binary in the static scrape config, which we could use instead of putting the namespace here. @demmer, any new thoughts?

The plugin here actually does two things:

1. Install the "/metrics" handler.
1. Set the namespace.

Would it make sense to move 1. to our servenv instead and install the handler by default? What are the downsides of having the handler in there by default?

re: 2. If setting the namespace is a crucial aspect, I would prefer to do it in code. This way it's more reusable. Otherwise, other people will ask you for this static map config ;) If you set the namespace in code, my comment is that you should set the namespace for all binaries, not just vtgate and vttablet.

Reply:
Oh interesting. 

1. I think moving it into servenv makes sense, location-wise. The thought here was that we wanted to be able to not have any Prometheus-stuff running if a build didn't want it, but I think we can handle that inside servenv with a flag. 
2. Yeah, setting the namespace is crucial. I've added one for the `vtworker` and `vtctl` binaries now, are there any that I missed that metrics are in too? Those are all the binaries that we use at Slack.